multiaddr_test.go

```diff
@@ -157,6 +158,8 @@ func TestConstructSucceeds(t *testing.T) {
 		"/ip4/127.0.0.1/tcp/1234",
 		"/ip4/127.0.0.1/tcp/1234/",
 		"/ip4/127.0.0.1/udp/1234/quic",
+		"/ip4/127.0.0.1/udp/1234/quic/webtransport",
+		"/ip4/127.0.0.1/udp/1234/quic/webtransport/certhash/zt1Zv2yaZ",
```

Comment:
IIUC there can be multiple certhashes, if so we should probably test for multiple certhashes as well.

I'm not sure if the place is here or elsewhere, but wherever we're defining the semantics of `webtransport` and `certhash` we should add some tests around escaping (e.g. making sure you can do `.../webtransport/certhash/hash1/certhash/hash2/p2p/peerID` but not `.../webtransport/certhash/hash1/p2p/peerID/certhash/hash2`

Reply:
> IIUC there can be multiple certhashes, if so we should probably test for multiple certhashes as well.

Good point. Will add.

> I'm not sure if the place is here or elsewhere, but wherever we're defining the semantics of webtransport and certhash we should add some tests around escaping

It probably needs to live in the WebTransport implementation. Having one certhash, then something else, then another certhash can be a valid multiaddr (e.g. a circuit).